extensions/wikia/Wall/WallBaseController.class.php

```diff
@@ -92,36 +101,34 @@ public function index( $wallMessagesPerPage = null ) {
 			$wallMessagesPerPage = self::DEFAULT_MESSAGES_PER_PAGE;
 		}
 
-		$this->getThreads( $title, $page, $wallMessagesPerPage );
+		/* @var Wall wall */
+		$wall = $this->getWallForIndexPage( $title );
+		$wall->setMaxPerPage( $wallMessagesPerPage );
+		$wall->setSorting( $this->getSortingSelected() );
+
+		$threads = $wall->getThreads( $page );
+		$countComments = $wall->getThreadCount();
+
+		$renderUserTalkArchiveAnchor = $this->request->getVal( 'dontRenderUserTalkArchiveAnchor', false ) != true;
 
 		$this->response->setVal( 'type', 'Board' );
 		$this->response->setVal( 'showNewMessage', true );
 		$this->response->setVal( 'condenseMessage', true );
 
-		$this->response->setVal( 'renderUserTalkArchiveAnchor', $this->request->getVal( 'dontRenderUserTalkArchiveAnchor', false ) != true );
-
-		$greeting = Title::newFromText( $title->getText(), NS_USER_WALL_MESSAGE_GREETING );
-		$greetingText = '';
-
-		if ( !empty( $greeting ) && $greeting->exists() ) {
-			$article = new Article( $greeting );
-			$article->getParserOptions();
-			$article->mParserOptions->setIsPreview( true ); // create parser option
-			$article->mParserOptions->setEditSection( false );
-			$greetingText = $article->getParserOutput()->getText();
-		}
-		wfRunHooks( 'WallGreetingContent', [ &$greetingText ] ); // used by SWM to add messages to Wall in monobook
-		$this->response->setVal( 'greeting', $greetingText );
-
+		$this->response->setVal( 'wall', $wall );
+		$this->response->setVal( 'threads', $threads );
+		$this->response->setVal( 'renderUserTalkArchiveAnchor', $renderUserTalkArchiveAnchor );
+		$this->response->setVal( 'greeting', $this->getGreetingText( $title ) );
 		$this->response->setVal( 'sortingOptions', $this->getSortingOptions() );
 		$this->response->setVal( 'sortingSelected', $this->getSortingSelectedText() );
-		$this->response->setVal( 'title', $title );
-		$this->response->setVal( 'totalItems', $this->countComments );
+		$this->response->setVal( 'title', $this->wg->Title );
+		$this->response->setVal( 'countComments', $countComments );
+		$this->response->setVal( 'totalItems', $countComments );
 		$this->response->setVal( 'itemsPerPage', $wallMessagesPerPage );
-		$this->response->setVal( 'showPager', ( $this->countComments > $wallMessagesPerPage ) );
+		$this->response->setVal( 'showPager', ( $countComments > $wallMessagesPerPage ) );
 		$this->response->setVal( 'currentPage', $page );
 
-		Transaction::setSizeCategoryByDistributionOffset( $this->countComments, 0, self::DEFAULT_MESSAGES_PER_PAGE );
+		Transaction::setSizeCategoryByDistributionOffset( $countComments, 0, self::DEFAULT_MESSAGES_PER_PAGE );
 
 		// TODO: keep the varnish cache and do purging on post
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)